plugins/micromega/coq_micromega.ml

```diff
@@ -1678,6 +1759,18 @@ let rec abstract_wrt_formula f1 f2 =
 
 exception CsdpNotFound
 
+let fail_csdp_not_found () =
+  flush stdout;
+  let s =
+    " Skipping what remains of this tactic: the complexity of the \
+     goal requires the use of a specialized external tool called csdp. \n\n\
```

Comment:
The way all those reflexive tactics work is as follows:
* first the goal to prove is reified (meaning parsed) into an internal type (this step is unverified)
* then this reified goal can be read by some untrusted code, usually an OCaml plugin, that generates some witness (also an unverified step)
* the goal is converted, using the `change` tactic, to the semantics of the reified goal (first actual step of the proof)
* a theorem is applied that states that if some checker ran on the reified goal and the witness returns `true`, then the semantics holds, thus changing the goal to something like `checker reified_goal witness = true` (second step of the proof)
* then `vm_compute` turns the goal to `true = true` and `reflexivity` concludes the proof

Do you think such an explanation (with a small exampe) would be needed in the refman? It seems quite technical to me and I don't feel the user of the tactic needs to know what goes under the hood to use the tactic.

Reply:
If this is indeed for internal use only, then no need to document it.  If you document it, maybe stating the steps is overkill unless it can be done more succinctly.  Some of it could be implicit in an example (in particular IIUC, your last 3 bullet points).  I don't understand the details (yet).